sycl/include/syclcompat/math.hpp

```diff
@@ -2242,3 +2302,30 @@ inline constexpr unsigned extend_vcompare4_add(AT a, BT b, unsigned c,
 }
 
 } // namespace syclcompat
+
+// Specialize std::common_type for bfloat16
+// Semantics here match bfloat16.hpp operator overloads (all mixed type math
+// ops return bfloat16)
+// TODO(syclcompat-lib-reviewers) Move this to bfloat extension
+namespace std {
+template <> struct common_type<sycl::ext::oneapi::bfloat16> {
+  using type = sycl::ext::oneapi::bfloat16;
+};
+
+template <>
+struct common_type<sycl::ext::oneapi::bfloat16, sycl::ext::oneapi::bfloat16> {
+  using type = sycl::ext::oneapi::bfloat16;
+};
+
+template <typename T> struct common_type<sycl::ext::oneapi::bfloat16, T> {
+  using type = sycl::ext::oneapi::bfloat16; // std::common_type_t<float, T>;  //
```

Comment:
I don't understand the comments. `std::commone_type_t<float, sycl::ext::oneapi::bfloat16>` will be `bfloat16`, and won't be `float`, why do the comments say `bfloat16` will be promoted to `float`?